src/ice_shelf/MOM_ice_shelf_initialize.F90

```diff
@@ -36,23 +39,48 @@ subroutine initialize_ice_thickness(h_shelf, area_shelf_h, hmask, G, US, PF)
                                              !! partly or fully covered by an ice-shelf
   type(unit_scale_type), intent(in)    :: US !< A structure containing unit conversion factors
   type(param_file_type), intent(in)    :: PF !< A structure to parse for run-time parameters
+  logical, intent(in), optional        :: rotate_index !< If true, this is a rotation test
+  integer, intent(in), optional        :: turns !< Number of turns for rotation test
 
   integer :: i, j
   character(len=40)  :: mdl = "initialize_ice_thickness" ! This subroutine's name.
   character(len=200) :: config
+  logical :: rotate = .false.
+  real, allocatable, dimension(:,:) :: tmp1_2d ! Temporary array for storing ice shelf input data
+  real, allocatable, dimension(:,:) :: tmp2_2d ! Temporary array for storing ice shelf input data
+  real, allocatable, dimension(:,:) :: tmp3_2d ! Temporary array for storing ice shelf input data
 
   call get_param(PF, mdl, "ICE_PROFILE_CONFIG", config, &
                  "This specifies how the initial ice profile is specified. "//&
                  "Valid values are: CHANNEL, FILE, and USER.", &
                  fail_if_missing=.true.)
 
-  select case ( trim(config) )
-  case ("CHANNEL"); call initialize_ice_thickness_channel (h_shelf, area_shelf_h, hmask, G, US, PF)
-  case ("FILE");  call initialize_ice_thickness_from_file (h_shelf, area_shelf_h, hmask, G, US, PF)
-  case ("USER");  call USER_init_ice_thickness (h_shelf, area_shelf_h, hmask, G, US, PF)
-  case default ;  call MOM_error(FATAL,"MOM_initialize: "// &
-    "Unrecognized ice profile setup "//trim(config))
-  end select
+  if (PRESENT(rotate_index)) rotate=rotate_index
+
+  if (rotate) then
+     allocate(tmp1_2d(G_in%isd:G_in%ied,G_in%jsd:G_in%jed));tmp1_2d(:,:)=0.0
+     allocate(tmp2_2d(G_in%isd:G_in%ied,G_in%jsd:G_in%jed));tmp2_2d(:,:)=0.0
+     allocate(tmp3_2d(G_in%isd:G_in%ied,G_in%jsd:G_in%jed));tmp3_2d(:,:)=0.0
```

Comment:
`allocate_rotated_array` may be useful here.

Also, as mentioned in the generic comment, we may want to spin off the rotation of the ice thickness parameters into a separate `rotate_ice_thickness` function.